src/adagrams.js

```diff
@@ -1,15 +1,144 @@
+const letterPool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const playersHand = [];
+  const otherLetters = [];
+  const copyOfLetterPool = {...letterPool};
+  for (let letter in copyOfLetterPool) {
+    for (let i = 0; i < copyOfLetterPool[letter]; i++) {
+      otherLetters.push(letter);
+    }
+  }
+  for (let i = 0; i < 10; i++) {
+    let index = Math.floor(Math.random() * otherLetters.length);
+    // add to playersHand
+    playersHand.push(otherLetters[index]);
+    // remove from other letters
+    otherLetters.splice(index, 1);
+  }
+  return playersHand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  const lengthOfHand = input.length;
+  let lettersInHandCopy = [...lettersInHand];
+  for (let i = 0; i < lengthOfHand; i++) {
+    let letter = input[i];
```

Comment:
Looks like you don't reassign `letter` in the for loop so we can declare the variable with const